internal/services/mssqlmanagedinstance/mssql_managed_database_data_source.go

```diff
@@ -0,0 +1,184 @@
+					},
+
+					// WeekOfYear - The week of year to take the yearly backup in an ISO 8601 format.
+					"week_of_year": {
+						Type:     pluginsdk.TypeInt,
+						Optional: true,
+						Computed: true,
+					},
+
+					"immutable_backups_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Computed: true,
+					},
+				},
+			},
+		},
+		// "long_term_retention_policy": helper.LongTermRetentionPolicySchema(),
+		"short_term_retention_days": {
+			Type:     pluginsdk.TypeInt,
+			Computed: true,
+		},
+		"point_in_time_restore": {
+			Type:     schema.TypeList,
+			Computed: true,
+		},
+	}
+}
+
+func (d MsSqlManagedDatabaseDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MSSQLManagedInstance.ManagedDatabasesClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			longTermRetentionClient := metadata.Client.MSSQLManagedInstance.ManagedInstancesLongTermRetentionPoliciesClient
+			shortTermRetentionClient := metadata.Client.MSSQLManagedInstance.ManagedInstancesShortTermRetentionPoliciesClient
+
+			var state MsSqlManagedDatabaseDataSourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding: %+v)", err)
+			}
+
+			managedInstanceId := parse.NewManagedInstanceID(subscriptionId, state.ResourceGroupName, state.ManagedInstanceName)
+			id := parse.NewManagedDatabaseID(subscriptionId, managedInstanceId.ResourceGroup, managedInstanceId.Name, state.Name)
+			resp, err := client.Get(ctx, id.ResourceGroup, id.ManagedInstanceName, id.DatabaseName)
+			if err != nil {
+				if utils.ResponseWasNotFound(resp.Response) {
+					return fmt.Errorf("%s was not found", id)
+				}
+				return fmt.Errorf("retrieving %s: %v", id, err)
+			}
+
+			model := MsSqlManagedDatabaseDataSourceModel{
+				Name:                id.DatabaseName,
+				ManagedInstanceName: managedInstanceId.Name,
+				ResourceGroupName:   id.ResourceGroup,
+			}
+
+			model.ManagedInstanceId = managedInstanceId.ID()
+
+			ltrResp, err := longTermRetentionClient.Get(ctx, id.ResourceGroup, id.ManagedInstanceName, id.DatabaseName)
+			if err != nil {
+				return fmt.Errorf("retrieving Long Term Retention Policy for  %s: %v", id, err)
+			}
+
+			model.LongTermRetentionPolicy = flattenLongTermRetentionPolicy(ltrResp)
+
+			shortTermRetentionResp, err := shortTermRetentionClient.Get(ctx, id.ResourceGroup, id.ManagedInstanceName, id.DatabaseName)
+			if err != nil {
+				return fmt.Errorf("retrieving Short Term Retention Policy for  %s: %v", id, err)
+			}
+
+			if shortTermRetentionResp.RetentionDays != nil {
+				model.ShortTermRetentionDays = int64(*shortTermRetentionResp.RetentionDays)
+			}
+
+			d := metadata.ResourceData
+			if v, ok := d.GetOk("point_in_time_restore"); ok {
+				model.PointInTimeRestore = flattenManagedDatabasePointInTimeRestore(v)
+			}
```

Comment:
I can see that this was copied from the corresponding resource, but I don't see a need for assigning this to a variable

```suggestion
			if v, ok := metadata.ResourceData.GetOk("point_in_time_restore"); ok {
				model.PointInTimeRestore = flattenManagedDatabasePointInTimeRestore(v)
			}
```